cmd/kubectl/kubectlcobra/grouping.go

```diff
@@ -211,3 +219,31 @@ func mapKeysToSlice(m map[string]string) []string {
 	}
 	return s
 }
+
+// addSuffixToName adds the passed suffix (usually a hash) as a suffix
+// to the name of the passed object stored in the Info struct. Returns
+// an error if the object is not "*unstructured.Unstructured" or if the
+// name stored in the object differs from the name in the Info struct.
+func addSuffixToName(info *resource.Info, suffix string) error {
+
+	if info == nil {
+		return fmt.Errorf("Nil resource.Info")
+	}
+	if len(suffix) == 0 {
```

Comment:
i'd say error here - swallowing this could lead to bugs.
The caller can test len(suffix) to avoid the error.